engine/src/main/java/org/hibernate/search/spi/SearchIntegratorBuilder.java

```diff
@@ -571,16 +572,28 @@ private static String defineIndexingStrategy(SearchConfiguration cfg) {
 	}
 
 	private ErrorHandler createErrorHandler(SearchConfiguration searchConfiguration) {
-		String errorHandlerClassName = searchConfiguration.getProperty( Environment.ERROR_HANDLER );
-		if ( StringHelper.isEmpty( errorHandlerClassName ) ) {
+		Object configuredErrorHandler = searchConfiguration.getProperties().get( Environment.ERROR_HANDLER );
+
+		if ( configuredErrorHandler == null ) {
 			return new LogErrorHandler();
 		}
-		else if ( "log".equals( errorHandlerClassName.trim() ) ) {
+		if ( configuredErrorHandler instanceof String ) {
+			return createErrorHandlerFromString( (String) configuredErrorHandler, searchConfiguration.getClassLoaderService() );
+		}
+		else if ( configuredErrorHandler instanceof ErrorHandler ) {
+			return (ErrorHandler) configuredErrorHandler;
+		}
+		else {
+			throw log.unsupportedErrorHandlerConfigurationValueType( configuredErrorHandler.getClass() );
+		}
+	}
+
+	private ErrorHandler createErrorHandlerFromString(String errorHandlerClassName, ClassLoaderService classLoaderService) {
+		if ( StringHelper.isEmpty( errorHandlerClassName ) || "log".equals( errorHandlerClassName.trim() ) ) {
```

Comment:
Should 'log' maybe be a constant, making it clear that it is the shortcut name of the logging error handler?